core/src/main/java/org/mule/runtime/core/internal/context/thread/notification/DefaultThreadNotificationService.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.context.thread.notification;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.lifecycle.InitialisationException;
+import org.mule.runtime.api.lifecycle.Lifecycle;
+import org.mule.runtime.api.scheduler.Scheduler;
+import org.mule.runtime.api.scheduler.SchedulerService;
+import org.mule.runtime.api.util.Pair;
+import org.mule.runtime.core.api.context.thread.notification.ThreadNotificationService;
+
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.function.Function;
+
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.apache.commons.lang3.StringUtils.leftPad;
+
+/**
+ * Implementation for {@link ThreadNotificationService}. It collects {@link ThreadNotificationElement}' and
+ * every X seconds logs statistics from them, using the {@link ThreadsStatistics}.
+ *
+ * @since 4.2
+ */
+public class DefaultThreadNotificationService implements ThreadNotificationService, Lifecycle {
+
+  private static final int DEFAULT_INTERVAL = 5;
```

Comment:
make this configurable via system property

Reply:
and provide a way to collect metrics but not log, so we can rely on an operation in the test plugin rather than the log